src/main/java/com/pkb/unit/Unit.java

```diff
@@ -36,7 +37,9 @@
 import io.vavr.collection.List;
 
 /**
- * A Unit is a base class for a restartable,
+ * A Unit is a base class for a restartable, stoppable service that can have depencencies and
+ * that other Unitscan depend on. When a Unit is started then it tries the start its dependencies.
```

Comment:
May I suggest something on top of @MFAshby's suggestion? :)
 	```suggestion
 * that other Units can depend on. When a Unit is started then it tries to start its dependencies.
```